File: Backend/src/test/kotlin/demo/backend/controller/WeatherForecastControllerTest.kt
package demo.backend.controller

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import demo.controller.WeatherForecastController
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.get
import java.time.LocalDateTime

@WebMvcTest(WeatherForecastController::class)
class WeatherForecastControllerTest(@Autowired private val mockMvc: MockMvc) {

    private val objectMapper = ObjectMapper().registerModule(JavaTimeModule())

    @Test
    fun weatherForecastControllerTest() {
        val result = mockMvc.get("/weatherforecast")
            .andExpect {
                status { isOk() }
            }.andReturn()

        val jsonString = result.response.contentAsString
        val resultArray = objectMapper.readValue(jsonString, Array<WeatherForecastController.WeatherForecast>::class.java)

        val now = LocalDateTime.now()
        Assertions.assertEquals(5, resultArray.size)
        for (forecastEntry in resultArray) {
            // forecast should always be in future
            Assertions.assertTrue(forecastEntry.date.isAfter(now))
            // Fahrenheit is always greater than Celsius
            Assertions.assertTrue(forecastEntry.temperatureC < forecastEntry.temperatureF)
        }
    }

}

File: Backend.Tests/Controllers/WeatherForecastControllerTest.cs
using Backend.Controllers;

namespace Backend.Tests.Controllers;

public class WeatherForecastControllerTest
{
    private WeatherForecastController _controller;

    [SetUp]
    public void Setup()
    {
        _controller = new WeatherForecastController();
    }

    [Test]
    public void TestGetWeatherForecast()
    {
        IEnumerable<WeatherForecast> result = _controller.Get();
        var resultList = result.ToList();

        Assert.That(resultList, Is.Not.Null);
        Assert.That(resultList.Count, Is.EqualTo(5));

        var now = DateTime.Now;

        foreach (var forecastEntry in resultList)
        {
            Assert.That(forecastEntry.Date > now, Is.True, "Forecast date must be in the future");
            Assert.That(forecastEntry.TemperatureC < forecastEntry.TemperatureF, Is.True, "Fahrenheit must be greater than Celsius");
        }
    }
}